Add placeholder for prices in PlanFeaturesHeader component

nitrajka opened the pull request:
There are some ways to fix this #8203. 

Do we want to display placeholders only for prices (in the square)? Or also for billing timeframe (in the oval)? 
![imageedit_20_7094482682](https://cloud.githubusercontent.com/assets/14272775/19050564/6d81e834-89af-11e6-91e6-c9b37ded27ee.jpg) 
In my PR, I'm displaying placeholders for prices both on jetpack plans site and the other plans site. And placeholder for timeframe on the normal plans site. 
Those placeholders are height enough to make `<PlanFeaturesHeader>`'s height the same while and after price loading. 

There is a difference between price height on jetpack plans site and price height on the other plans site. So I added there a condition when displaying a placeholder for a price. And according to this condition made styles for jetpack price placeholder and for the original site placeholders.

gwwar commented:
> Do we want to display placeholders only for prices (in the square)? Or also for billing timeframe (in the oval)? 

From https://github.com/Automattic/wp-calypso/pull/6588#issuecomment-231381945 it looks like we had one for each. @lamosty do you recall if we ever changed our minds on that?

lamosty commented:
> From #6588 (comment) it looks like we had one for each. @lamosty do you recall if we ever changed our minds on that?

Looks like it was changed in #7402 by @tyxla. I think we still want the placeholder over the billing timeframe as well.

Could you add them here? @nitrajka

gwwar commented:
Thanks @nitrajka for looking at this! 

I tested the branch and it works well. @lamosty did you not see the timeline placeholder? I can see it while running the branch.

Some of these placeholders look a bit big/heavy to my developer 👀 maybe @mtias or @adambbecker have a few width/height suggestions?

lamosty commented:
>  @lamosty did you not see the timeline placeholder? I can see it while running the branch.

Oh, I can see it too. I didn't test the branch, just replied. :)

> I tested the branch and it works well.

Working well for me, too.

> Some of these placeholders look a bit big/heavy [...]

Yeah, maybe the first one could be a bit shorter:

![selection_019](https://cloud.githubusercontent.com/assets/4988512/19072680/d7903bc2-8a36-11e6-8375-6fbb5283b57a.jpg)

but best would be if some designer took a look at it.

nitrajka commented:
I refactored the condition according to the @gwwar's suggestion and also added a breakpoint to the timeframe placeholder. Without a breakpoint, on small screens (like mobiles) the height of `<PlanFeaturesHeader>` while and after loading wasn't the same.

I tested those placeholders and also their responsivity in Firefox, Chrome and Safari. It works well for me.

What about some tests? Maybe I could add some to test the placeholders.

gwwar commented on client/my-sites/plan-features/header.jsx:
```diff
@@ -147,12 +147,19 @@ class PlanFeaturesHeader extends Component {
+		const classes = classNames( 'is-placeholder', {
+			'plan-features__price': ! isJetpackSite,
+			'plan-features__price-jetpack': isJetpackSite
+		} );
```
Let's move this into the`if ( isPlaceholder )` block

gwwar commented:
> What about some tests? Maybe I could add some to test the placeholders.

No need for tests in this case. Thanks though 😄 If you're interested in writing a few component tests, we have some issues open here for our more commonly used components https://github.com/Automattic/wp-calypso/milestone/125

gifs of branch behavior:

![wpcom](https://cloud.githubusercontent.com/assets/1270189/19088526/6aaa8cb6-8a2b-11e6-8c97-c37c9373cc1e.gif)
![jetpack](https://cloud.githubusercontent.com/assets/1270189/19088539/75cb455e-8a2b-11e6-83b2-6dd7d61600d9.gif)

Thanks a ton @nitrajka for the contribution. I think the behavior feels good enough to 🚢 We can always iterate in another PR if not.

nitrajka commented:
Thanks @gwwar , I'm glad I could help :) Sometimes it is not as easy to select an issue because it is already assigned to sb or while working on it, sb makes a PR 😃 thank you for suggesting another issue 😄

gwwar commented:
Thank you @nitrajka ! If you have something in particular you'd like to learn more about, I'm sure we can find some open issues for you.

Changes are now out in prod.
![deployed](https://cloud.githubusercontent.com/assets/1270189/19089537/88072cf2-8a2f-11e6-9269-11a9296c968f.gif)
